ocs_ci/ocs/warp.py

```diff
@@ -0,0 +1,149 @@
+        log.info(f"Installing warp S3 benchmark on testing pod {self.pod_obj.name}")
+        warp_path = "https://github.com/minio/warp/releases/download/v0.5.5/warp_0.5.5_Linux_x86_64.tar.gz"
+        self.pod_obj.exec_cmd_on_pod(f"wget {warp_path}", timeout=timeout)
+        self.pod_obj.exec_cmd_on_pod(
+            "tar -xzvf warp_0.5.5_Linux_x86_64.tar.gz", timeout=timeout
+        )
+        log.info("Successfully installing warp s3 benchmark")
+
+    def run_benchmark(
+        self,
+        bucket_name=None,
+        access_key=None,
+        secret_key=None,
+        duration=None,
+        concurrent=None,
+        objects=None,
+        obj_size=None,
+        timeout=None,
+    ):
+        """
+         Running Warp S3 benchmark
+         Usage detail can be found at: https://github.com/minio/warp
+
+        Args:
+            bucket_name (string): Name of bucket
+            access_key (string): Access Key credential
+            secret_key (string): Secret Key credential
+            duration (string): Duration of the test
+            concurrent (int): number of concurrent
+            objects (int): number of objects
+            obj_size (int): size of object
+            timeout (int): timeout in seconds
+
+        """
+
+        # Running warp S3 benchmark
+        log.info("Running Minio Warp S3 benchmark")
+        timeout = timeout if timeout else 3600
+        self.access_key = access_key if access_key else self.access_key
+        self.secret_key = secret_key if secret_key else self.secret_key
+        self.bucket_name = bucket_name if bucket_name else self.bucket_name
+        self.duration = duration if duration else self.duration["duration"]
+        self.concurrent = concurrent if concurrent else self.concurrent["concurrent"]
+        self.objects = objects if objects else self.objects["objects"]
+        self.obj_size = obj_size if obj_size else self.obj_size["obj.size"]
+        self.pod_obj.exec_cmd_on_pod(
+            f"{self.warp_bin_dir} mixed "
+            f"--duration={self.duration} "
+            f"--host={self.host} "
+            f"--access-key={self.access_key} "
+            f"--secret-key={self.secret_key} "
+            f"--noclear --noprefix --concurrent={self.concurrent} "
+            f"--objects={self.objects} "
+            f"--obj.size={self.obj_size} "
+            f"--get-distrib={self.get_distrib} "
+            f"--put-distrib={self.put_distrib} "
+            f"--delete-distrib={self.delete_distrib} "
+            f"--stat-distrib={self.stat_distrib} "
+            f"--analyze.v --bucket={self.bucket_name}",
+            out_yaml_format=False,
+            timeout=timeout,
+        )
+
+    def cleanup(self):
+        """
+        Clear all objects in the associated bucket
+        Clean up deployment config, pvc, pod and test user
+
+        """
+        log.info("Deleting pods and deployment config")
+        run_cmd(f"oc delete deploymentconfig/{self.pod_name} -n {self.namespace}")
```

Comment:
You can use https://github.com/red-hat-storage/ocs-ci/blob/55de5d72da977055339e07a4334759786823c121/ocs_ci/ocs/resources/pod.py#L1784
And then you will also have verification that the pod was deleted.

Reply:
Updated the code to use delete_deploymentconfig_pods(pod_obj) instead.